lib/internal/streams/web.js

```diff
@@ -0,0 +1,4632 @@
+
+    if (`${mode}` !== 'byob')
+      throw new ERR_INVALID_ARG_VALUE('options.mode', mode);
+    return new ReadableStreamBYOBReader(this);
+  }
+
+  /**
+   * @param {ReadableWritablePair} transform
+   * @param {StreamPipeOptions} [options]
+   * @returns {ReadableStream}
+   */
+  pipeThrough(transform, options) {
+    if (!isReadableStream(this))
+      throw new ERR_INVALID_THIS('ReadableStream');
+    const readable = transform?.readable;
+    if (!isReadableStream(readable)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'transform.readable',
+        'ReadableStream',
+        readable);
+    }
+    const writable = transform?.writable;
+    if (!isWritableStream(writable)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'transform.writable',
+        'WritableStream',
+        writable);
+    }
+
+    // The web platform tests require that these be handled one at a
+    // time and in a specific order. options can be null or undefined.
+    const preventAbort = options?.preventAbort;
+    const preventCancel = options?.preventCancel;
+    const preventClose = options?.preventClose;
+    const signal = options?.signal;
+
+    if (signal !== undefined && signal?.[kAborted] === undefined)
+      throw new ERR_INVALID_ARG_TYPE('options.signal', 'AbortSignal', signal);
+
+    if (isReadableStreamLocked(this))
+      throw new ERR_INVALID_STATE.TypeError('The ReadableStream is locked');
+    if (isWritableStreamLocked(writable))
+      throw new ERR_INVALID_STATE.TypeError('The WritableStream is locked');
+
+    const promise = readableStreamPipeTo(
+      this,
+      writable,
+      !!(preventClose),
+      !!(preventAbort),
+      !!(preventCancel),
+      signal);
+    setPromiseHandled(promise);
+
+    return readable;
+  }
+
+  /**
+   * @param {WritableStream} destination
+   * @param {StreamPipeOptions} [options]
+   * @returns {Promise<void>}
+   */
+  pipeTo(destination, options = {}) {
+    try {
+      if (!isReadableStream(this))
+        throw new ERR_INVALID_THIS('ReadableStream');
+      if (!isWritableStream(destination)) {
+        throw new ERR_INVALID_ARG_TYPE(
+          'transform.writable',
+          'WritableStream',
+          destination);
+      }
+
+      const preventAbort = options?.preventAbort;
+      const preventCancel = options?.preventCancel;
+      const preventClose = options?.preventClose;
+      const signal = options?.signal;
+
+      if (signal !== undefined && signal?.[kAborted] === undefined)
+        throw new ERR_INVALID_ARG_TYPE('options.signal', 'AbortSignal', signal);
+
+      if (isReadableStreamLocked(this))
+        throw new ERR_INVALID_STATE.TypeError('The ReadableStream is locked');
+      if (isWritableStreamLocked(destination))
+        throw new ERR_INVALID_STATE.TypeError('The WritableStream is locked');
+
+      return readableStreamPipeTo(
+        this,
+        destination,
+        !!(preventClose),
+        !!(preventAbort),
+        !!(preventCancel),
```

Comment:
are these extra parens a node convention?
```suggestion
        !!preventClose,
        !!preventAbort,
        !!preventCancel,
```

Reply:
They're a James Convention ;-) No problem removing them.